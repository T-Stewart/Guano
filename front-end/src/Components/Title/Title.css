.title-container {
  background-color: transparent;
  margin-left: 50px;
  margin-right: 50px;
}

.header-text {
  text-align: right;
  color: #bfc3ce;
  margin-right: 20px;
  font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
    "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
}

.guano {
  text-align: right;
  margin-right: 20px;
  padding-left: 100px;
  color: #767c91;
  font-family: monospace;
}

@media screen and (max-width: 768px) {
  .header-text,
  .opinion-header {
    text-align: center;
    color: #bfc3ce;
    font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
      "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
    margin-right: 0px;
    margin-left: 0px;
  }
  .guano {
    text-align: center;
    margin-right: 20px;
    color: #767c91;
    font-family: monospace;
    font-size: 500%;
    margin-right: 0px;
    padding: 0px;
  }
}
