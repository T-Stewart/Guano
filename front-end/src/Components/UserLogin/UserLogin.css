.wrapper {
  background: #25272e;
  margin-top: 25px;
  margin-left: 25px;
  margin-right: 25px;
  height: 700px;
  border-radius: 25em;
  display: grid;
  justify-content: center;
}

.login-title {
  text-align: center;
  color: #cde9ff;
  text-decoration: underline;
  font-family: monospace;
  margin-top: 50%;
  margin-bottom: -50%;
}

.login-form-group {
  display: grid;
}

.label {
  color: #cde9ff;
  font-family: "Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
    "Lucida Sans Unicode", Geneva, Verdana, sans-serif;
  font-weight: normal;
  font-size: large;
  text-align: left;
}

.login-btn {
  width: 150px;
  height: 50px;
  font-size: 150%;
  color: #24262d;
  margin-bottom: 10px;
  border-radius: 32px;
  background-color: #9ca6c9;
  font-family: monospace;
  text-align: center;
  cursor: pointer;
  transition: all 0.1s ease-in-out;
}
