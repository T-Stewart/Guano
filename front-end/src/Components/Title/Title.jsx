import React from "react"
import './Title.css'

export default class Title extends React.Component{
    render(){
        return(
      <div className="title-container" data-testid="title-test">
        <h1 className="guano">Guano</h1>
        <h2 className="header-text">Your Opinion Does Not Matter</h2>
        <h4 className="header-text"> But tell us anyway </h4>
      </div>
       );
    };
};