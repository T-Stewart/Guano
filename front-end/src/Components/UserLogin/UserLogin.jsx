import React from "react";
import axios from "axios";
import Title from "../Title/Title"
import "./UserLogin.css"

export default class UserLogin extends React.Component{

    constructor(props){
        super(props)
        this.onChangeEmail = this.onChangeEmail.bind(this)
        this.onChangePassword= this.onChangePassword.bind(this)
        this.onSubmit = this.onSubmit.bind(this)
        this.state = {
            email: '', 
            password: ''
        };
    };

    onChangeEmail = (e) => {
        this.setState({
            email: e.target.value
        });
    };

    onChangePassword = (e) => {
        this.setState({
            password: e.target.value
        });
    };

    onSubmit = (e) => {
        e.preventDefault();
        let wrongPassword = document.getElementById('wrong-password')    
        const user = {
            email: this.state.email,
            password: this.state.password
        };

        axios ({
            url: '/api/user/login',
            method: 'POST', 
            data: user
        })
        .then(response => {
            console.log('Data has been sent to server now');
            console.log(response)
            if(response.data.success){
                window.location.replace("/posts/" + response.data.userId)
            }else{
                wrongPassword.innerHTML = "Email and/or password is incorrect, please try again"
                wrongPassword.setAttribute("style", "color:red");
            }
        })
        .catch(err =>{
            console.log(err)
        });
    };

    showPassword() {
        let passwordInput = document.getElementById("password-id");
        if (passwordInput.type === "password") {
            passwordInput.type = "text";
        } else {
            passwordInput.type = "password"
        };
    };

    render() {
        return(
            <div>
                <Title/> 
                <div className="wrapper">                
                    <h3 className="login-title">Log into Your Account</h3>
                    <form className="login-form-group" onSubmit={this.onSubmit}>
                        <div >
                            <label className="label">Email</label>
                            <input 
                                type="email"
                                placeholder= "Email"
                                required
                                id="email-id"
                                className="form-control"
                                value={this.state.email}
                                onChange={this.onChangeEmail} />
                            <label className="label">Password</label> <div id="wrong-password"></div>
                            <input 
                                type="password" 
                                placeholder="Password"
                                required
                                id="password-id"
                                className='form-control'
                                value={this.state.password}
                                onChange={this.onChangePassword} />
                            
                            <div className="label">
                                <input type="checkbox" onChange={this.showPassword}/>
                                <label className="see-pass-wording">Show Password</label>
                            </div>
                        </div>
                        <div className="form-group">
                            <input type="submit" id="login" value="Login" className="login-btn"/>
                        </div>
                    </form>
                </div>
            </div>
           
        )
    }
}