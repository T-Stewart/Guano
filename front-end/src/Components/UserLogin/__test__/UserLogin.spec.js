import React from 'react';
import { shallow, mount } from 'enzyme';
import UserLogin from '../UserLogin'
import Title from '../../Title/Title'

describe('UserLogin', () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<UserLogin />));
    it('Should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('should render the Title component correctly', () => {
    expect(wrapper.containsAllMatchingElements([<Title />])).toEqual(true)
    })

    it('should render a <Div />', () => {
        expect(wrapper.find('div').length).toEqual(6)
    })

})

describe('mountedUserLogin', () => {
    let wrapper;

    beforeEach(() => wrapper = mount(<UserLogin />));
    
    it('calls onSubmit when submit button is clicked', () => {
        const spy = jest.spyOn(wrapper.instance(), 'onSubmit');
        wrapper.instance().forceUpdate();
        expect(spy).toHaveBeenCalledTimes(0);
        wrapper.find('.login-btn').first().simulate('submit');
        expect(spy).toHaveBeenCalledTimes(1);
    })
})
