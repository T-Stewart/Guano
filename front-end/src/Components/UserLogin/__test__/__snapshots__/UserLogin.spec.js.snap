// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLogin Should render correctly 1`] = `ShallowWrapper {}`;
